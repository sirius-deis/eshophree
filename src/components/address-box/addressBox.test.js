import { render, screen , fireEvent} from '@testing-library/react';
import AddressBox from './addressBox';

describe('AddressBox Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<AddressBox />);
    expect(container).toMatchSnapshot();
  });
  it('should render all label inputs correctly', () => {
    render(<AddressBox />);

    expect(screen.getByText(/First Name/)).toBeInTheDocument();
    expect(screen.getByText(/Last Name/)).toBeInTheDocument();
    expect(screen.getByText(/Street/)).toBeInTheDocument();
    expect(screen.getByText(/City/)).toBeInTheDocument();
    expect(screen.getByText(/State/)).toBeInTheDocument();
    expect(screen.getByText(/Zip Code/)).toBeInTheDocument();
    expect(screen.getByText(/Country/)).toBeInTheDocument();
    expect(screen.getByText(/Phone Number/)).toBeInTheDocument();
    expect(screen.getByText(/Email Address/)).toBeInTheDocument();
  });
  it('should update input value on change event', () => {
    render(<AddressBox />);
    
    const inputs = [
      { label: /First Name/, value: 'John' },
      { label: /Last Name/, value: 'Doe' },
      { label: /Street/, value: '123 Main St' },
      { label: /City/, value: 'Anytown' },
      { label: /State/, value: 'CA' },
      { label: /Zip Code/, value: '12345' },
      { label: /Country/, value: 'USA' },
      { label: /Phone Number/, value: '123-456-7890' },
      { label: /Email Address/, value: 'john.doe@example.com' }
    ];

    inputs.forEach(({ label, value }) => {
      const inputElement = screen.getByLabelText(label);
      fireEvent.change(inputElement, { target: { value } });
      expect(inputElement.value).toBe(value);
    });
  });
});