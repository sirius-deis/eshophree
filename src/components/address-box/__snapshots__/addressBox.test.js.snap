// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBox Component should match snapshot 1`] = `
<div>
  <div>
    <div
      class="sc-fFGjHI eYNeRf"
    >
      <div
        class="sc-dicizt iXKfqv"
      >
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            First Name
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="firstName"
              placeholder="Enter your First Name"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            Last Name
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="lastName"
              placeholder="Enter your Last Name"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-fFGjHI eYNeRf"
    >
      <div
        class="sc-dicizt iXKfqv"
      >
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            Street
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="street"
              placeholder="Enter your Street"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            City
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="city"
              placeholder="Enter your City"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            State
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="state"
              placeholder="Enter your State"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            Zip Code
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="zipCode"
              placeholder="Enter your Zip Code"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            Country
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="country"
              placeholder="Enter your Country"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-fFGjHI eYNeRf"
    >
      <div
        class="sc-dicizt iXKfqv"
      >
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            Phone Number
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="phone"
              placeholder="Enter your Phone Number"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="sc-bgqQcB cyUqvl"
        >
          <label
            class="sc-gTRrQi fYROVA"
          >
            Email Address
             *
            <input
              class="sc-ewnqHT bzVbyC"
              name="email"
              placeholder="Enter your Email Address"
              required=""
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
